### PR TITLE
Make db connect more robust

### PR DESCRIPTION
**Pull request by pbrackin**
Repair scoping issue which prevents the ORM errors from showing up in the log.

### CODE REVIEW

**Comment by zeripath** [comment]
I think you need to rebase on to master and just cherry pick your latest change.

**Comment by pbrackin** [comment]
Yeah what I did was rebase locally, but then I forgot to push the rebase to origin, so I just did the manual merge on the origin here.  At least I think that is what I did.

**Comment by pbrackin** [comment]
ok yeah the other thing I inserted to test it all out was an intermediate error holder & pass that out so as to make an info log print showing any intermediate failures at the info log level.  I was going to remove that but it seemed like it might be helpful.  Its complaining becz the naming convention right now.  You like this idea, or I can just take it out alltogether?

File: routers/init.go [hunk 1]
```diff
@@ -44,21 +44,25 @@ func NewServices() {
...
 // In case of problems connecting to DB, retry connection. Eg, PGSQL in Docker Container on Synology
-func initDBEngine() (err error) {
+
+func initDBEngine() (err error, terminal bool) {
 	log.Info("Beginning ORM engine initialization.")
+	var im_err error
 	for i := 0; i < setting.DBConnectRetries; i++ {
 		log.Info("ORM engine initialization attempt #%d/%d...", i+1, setting.DBConnectRetries)
-		if err := models.NewEngine(migrations.Migrate); err == nil {
+		if err = models.NewEngine(migrations.Migrate); err == nil {
```
Comment (zeripath) [inline routers/init.go:51]: This is the important line

**Review by zeripath — CHANGES_REQUESTED** [review]
Only the line I've commented on should be changed. I don't believe the rest of these changes are necessary - and in fact are likely to lead to more bugs.

**Comment by zeripath** [comment]
Take em out.

File: routers/init.go (hunk 1 above)
Comment (pbrackin) [reply routers/init.go:51]: Correct.  I left the additional change to give you a chance to take or leave.  It just adds the showing - at the default info log level - the last intermediate failures such that someone does not need to set log level to debug to see that there was an intermediate failure upon ultimate success.  I will rewind these for you.

Comment (zeripath) [reply routers/init.go:51]: It's too complex I think... Just feels messy.

**Review by techknowlogick — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Pull request merged**